engine/src/main/scala/cromwell/engine/workflow/lifecycle/execution/job/EngineJobExecutionActor.scala

```diff
@@ -208,18 +206,21 @@ class EngineJobExecutionActor(replyTo: ActorRef,
   }
 
   when(FetchingCachedOutputsFromDatabase) {
-    case Event(CachedOutputLookupSucceeded(womValueSimpletons, jobDetritus, returnCode, cacheResultId, cacheHitDetails), data: ResponsePendingData) =>
-      writeToMetadata(Map(
-        callCachingHitResultMetadataKey -> true,
-        callCachingReadResultMetadataKey -> s"Cache Hit: $cacheHitDetails"))
-      log.debug("Cache hit for {}! Fetching cached result {}", jobTag, cacheResultId)
-      makeBackendCopyCacheHit(womValueSimpletons, jobDetritus, returnCode, data, cacheResultId) using data.withCacheDetails(cacheHitDetails)
+    case Event(CachedOutputLookupSucceeded(womValueSimpletons, jobDetritus, returnCode, cacheResultId, cacheHitDetails), data @ ResponsePendingData(_, _, _, _, Some(ejeaCacheHit), _)) =>
+      if (cacheResultId != ejeaCacheHit.hit.cacheResultId) {
+        // Sanity check: was this the right set of results (a false here is a BAD thing!):
+        log.error(s"Received incorrect call cache results from FetchCachedResultsActor. Expected ${ejeaCacheHit.hit} but got $cacheResultId. Running job")
+        // Treat this like the "CachedOutputLookupFailed" event:
+        runJob(data)
```

Comment:
Re "Treat this like the "CachedOutputLookupFailed" event", I wonder whether in both cases ` invalidateCacheHitAndTransition(cacheHit, data, throwable)` would be a more appropriate response than `runJob(data)`?

Reply:
Mainly because I noticed that we can rather than because we've noticed this going awry. If you want I can change this to be "write the error but don't change behavior from what it was before (ie use the results regardless)"?